### ISSUE TITLE
fastlane init fails with `full_gem_path': uninitialized constant Bundler::Plugin::API::Source (NameError)

### ISSUE THREAD

**Issue by rowets** [fastlane/fastlane#6203]
Hi guys

I'm new to fastlane and I've just been following the setup instructions to install and init fastlane, here are the commands I've run:

sudo gem install fastlane --verbose

I then go into the project directory which holds the ios .xcodeproj file and try to create the init file with the following:

fastlane init

However, it falls over straight away with the message below, does anybody know how I can get up and running (i only want to specify Xcode 8 for snapshot but it looks like i need to run it within snapshot and I can't get snapshot to even create the snapfile!)

TrevR

/Library/Ruby/Gems/2.0.0/gems/bundler-1.13.1/lib/bundler/rubygems_ext.rb:45:in `full_gem_path': uninitialized constant Bundler::Plugin::API::Source (NameError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1391:in `contains_requirable_file?'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:68:in `block in require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:823:in `block in each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `find'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.13.1/lib/bundler/plugin/api/source.rb:2:in `<top (required)>'
	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.13.1/lib/bundler/rubygems_ext.rb:45:in `full_gem_path'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1391:in `contains_requirable_file?'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:68:in `block in require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:823:in `block in each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `find'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.13.1/lib/bundler.rb:11:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.103.0/lib/fastlane/plugins/plugin_manager.rb:3:in `<class:PluginManager>'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.103.0/lib/fastlane/plugins/plugin_manager.rb:2:in `<module:Fastlane>'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.103.0/lib/fastlane/plugins/plugin_manager.rb:1:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.103.0/lib/fastlane/plugins/plugins.rb:9:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.103.0/lib/fastlane.rb:22:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.103.0/lib/fastlane/cli_tools_distributor.rb:14:in `take_off'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.103.0/bin/fastlane:5:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'

**Comment by wuaschtikus** [comment]
I just hit the same issue after I did a gem update

**Comment by ElChapitan** [comment]
I ran into the same problem, found a solution here:  https://github.com/fastlane/fastlane/issues/6066

**Comment by wuaschtikus** [comment]
Thank God! 
for me it was version 1.13.1

`gem uninstall bundler -v1.13.1`

**Comment by rowets** [comment]
Thanks, for me it was also 1.13.1...I can get on with learning fastlane now!

**Comment by 0xced** [comment]
This bundler issue is tracked on https://github.com/bundler/bundler/pull/4981

**Comment by 0xced** [comment]
And here's how to fix this error on Travis (works with any macOS image):
```yaml
before_install:
  # Force bundler 1.12.5 because version 1.13 has issues, see https://github.com/fastlane/fastlane/issues/6065#issuecomment-246044617
  - gem uninstall bundler -v '>1.12.5' --force || echo "bundler >1.12.5 is not installed"
  - gem install bundler -v 1.12.5 --no-rdoc --no-ri --no-document --quiet
```

**Comment by TKBurner** [comment]
@0xced Thanks so much for sharing!

**Comment by wkoszek** [comment]
@wuaschtikus @0xced same here, and your solution fixed the issue for my Travis build. Thanks a lot.

**Comment by wkoszek** [comment]
@TKBurner @wanbok @0xced This time I've hit this not only with Travis CI, but also with my Mac. The El Captain with XCode 8. Running `frameit` gives me:

```
frameit
/Users/wk/.rvm/gems/ruby-2.2.4@global/gems/bundler-1.13.1/lib/bundler/rubygems_ext.rb:45:in `full_gem_path': uninitialized constant Bundler::Plugin::API::Source (NameError)
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/basic_specification.rb:144:in `block in full_require_paths'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/basic_specification.rb:143:in `map'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/basic_specification.rb:143:in `full_require_paths'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/basic_specification.rb:76:in `contains_requirable_file?'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:949:in `block in find_in_unresolved'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:949:in `each'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:949:in `find_all'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:949:in `find_in_unresolved'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:74:in `require'
       	from /Users/wk/.rvm/gems/ruby-2.2.4@global/gems/bundler-1.13.1/lib/bundler/plugin/api/source.rb:2:in `<top (required)>'
       	from /Users/wk/.rvm/gems/ruby-2.2.4@global/gems/bundler-1.13.1/lib/bundler/rubygems_ext.rb:45:in `full_gem_path'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/basic_specification.rb:144:in `block in full_require_paths'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/basic_specification.rb:143:in `map'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/basic_specification.rb:143:in `full_require_paths'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1370:in `add_self_to_load_path'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1283:in `activate'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1314:in `block in activate_dependencies'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1300:in `each'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1300:in `activate_dependencies'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1282:in `activate'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1314:in `block in activate_dependencies'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1300:in `each'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1300:in `activate_dependencies'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/specification.rb:1282:in `activate'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_gem.rb:67:in `block in gem'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_gem.rb:66:in `synchronize'
       	from /Users/wk/.rvm/rubies/ruby-2.2.4/lib/ruby/site_ruby/2.2.0/rubygems/core_ext/kernel_gem.rb:66:in `gem'
       	from /Users/wk/.rvm/gems/ruby-2.2.4/bin/frameit:22:in `<main>'
       	from /Users/wk/.rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `eval'
       	from /Users/wk/.rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `<main>'
```

Did the suggested workaround, but it didn't work.

**Comment by wkoszek** [comment]
Actually the workaround worked fine. I've had to do this:

```
gem uninstall -i /Users/wk/.rvm/gems/ruby-2.2.4@global bundler
```

since I think something is messed up with my `GEM_HOME` variable.

**Comment by wanbok** [comment]
@wkoszek You must call @wuaschtikus , not me :p

**Comment by WaKeMaTTa** [comment]
# Solution

Check if you have the right version of RubyGems

```console
$ gem -v
2.6.5 (or superior)
```

If you have not, upgrade your RubyGems:

```console
$ gem update --system

if you can't use `sudo` like this:
$ sudo gem update --system
```

---

Now check you have the last bundler version:

```console
$ bundle -v
Bundler version 1.13.0 (or superior)
```

If you have not, upgrade your Bundler:

```
gem install bundler
```

**Comment by TKBurner** [comment]
@WaKeMaTTa Thanks so much for sharing this.

Can anyone let me know if they are still affected after this workaround? Thanks!

**Comment by msmollin** [comment]
@TKBurner - fixed for me. I only needed to do part 1 of @WaKeMaTTa's solution. I'm running 10.11.6 though so I don't know how this will hold up on 10.12 machines.

**Comment by WaKeMaTTa** [comment]
**PS**: im using ubuntu 16.04

**Comment by KrauseFx** [comment]
@WaKeMaTTa Sorry, Linux is not officially supported by _fastlane_ at that moment

**Comment by WaKeMaTTa** [comment]
@KrauseFx I know but what i want to say is this issue is not from your gem is from RubyGems and Bundler.

**Comment by pixelrevision** [comment]
Updating gems, cleaning and installing again worked for me macOS 10.12.

**Comment by TKBurner** [comment]
@pixelrevision Awesome! 👍

**Comment by 0xced** [comment]
It has been fixed in bundler 1.13.4, see https://github.com/bundler/bundler/pull/5010
